azure/ConsiderationsForServiceDesign.md

```diff
@@ -77,6 +77,71 @@ Understanding how your service is used and defining its model and interaction pa
 
 :white_check_mark: **DO** create an [OpenAPI Definition](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md) (with [autorest extensions](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md)) describing the service. The OpenAPI definition is a key element of the Azure SDK plan and is essential for documentation, usability and discoverability of services.
 
+## Use Good Names
+
+Good names for resources, properties, operations, and parameters are essential to a good developer experience.
+
+Resources are described by nouns. Resource and property names must be descriptive and easy for customers to understand. Ease of understanding comes from familiarity and recognition; you should favor consistency with other Azure services, names in the product's portal/user interface, and industry standards.
+
+Names should aid developers in discovering functionality without having to constantly refer to documentation.
+Use common patterns and standard conventions to aid developers in correctly guessing common property names and meanings.
+Use verbose naming patterns and avoid abbreviations other than
+well-known acronyms in your service domain.
+
```

Comment:
I think it would be useful to add something about a common naming mistake I see: API designers often name concepts based on how the concepts work (architecture and design), not what the concepts are for (the user scenario), i.e. they name them from the perspective of the concept designer, not the user. For example, AppConfig service has a concept KeyValuePair. A better name for the same concept is ConfigurationSetting, as what thew user has in mind when they work with configation is settings. The kev-value pair is one way to design a setting system